mempool/mempool.go

```diff
@@ -94,6 +98,12 @@ func NewMempool(config *cfg.MempoolConfig, proxyAppConn proxy.AppConnMempool) *M
 	return mempool
 }
 
+// FireOnTxsAvailable initializes the TxsAvailable channel,
+// ensuring it will trigger once every height when transactions are available.
+func (mem *Mempool) FireOnTxsAvailable() {
```

Comment:
Thanks for pointing this out. The thinking was that it should only be called once at startup time. We should change the comment doc to reflect that. In the future, if we decide we want to be able to toggle this behaviour, we'll need a mutex.